pkg/logs/internal/launchers/file/launcher.go

```diff
@@ -203,7 +233,11 @@ func (s *Launcher) removeSource(source *sources.LogSource) {
 
 // launch launches new tailers for a new source.
 func (s *Launcher) launchTailers(source *sources.LogSource) {
-	files, err := s.fileProvider.collectFiles(source)
+	// If we're at the limit already, no need to do a 'CollectFiles', just wait for the next 'scan'
```

Comment:
What if the new file has a more recent mtime than the ones we're tailing already?

Reply:
I think the only solution in that case would be to manually trigger a `scan` to consider all log sources.

If some file matching the new source has a newer mtime than the currently monitored set, we still need to re-evaluate _all_ matching file mtimes in order to determine what the top N are, what if some file matching a different source has an even newer mtime?

I decided against triggering a scan here and opted to let the next scheduled `scan` take care of this.